tee-worker/service/src/main.rs

```diff
@@ -694,25 +690,32 @@ fn start_worker<E, T, D, InitializationHandler, WorkerModeProvider>(
 	// ------------------------------------------------------------------------
 	// subscribe to events and react on firing
 	println!("*** Subscribing to events");
-	let (sender, receiver) = channel();
-	let metadata = node_api.metadata.clone();
-	let _ = thread::Builder::new()
-		.name("event_subscriber".to_owned())
-		.spawn(move || {
-			node_api.subscribe_events(sender).unwrap();
-		})
-		.unwrap();
-
-	println!("[+] Subscribed to events. waiting...");
-	let timeout = Duration::from_secs(600);
 	loop {
-		if let Ok(events_str) = receiver.recv_timeout(timeout) {
-			let event_bytes = Vec::from_hex(events_str).unwrap();
-			let events = Events::new(metadata.clone(), Default::default(), event_bytes);
+		let (sender, receiver) = channel();
```

Comment:
Same as above - I think we are fine with disruption without reconnection, it won't break the loop anyway and it just stops printing events which is not a big deal. We can add logs when disconnected though.

What do you think?

Reply:
Actually,  if just add add log when disconnected, it will not sync Event, is it ok? if ok, then just print error log. 

For me, I think  it's better to reconnect automatically, or the operator/user need to always check the log and restart the program to sync Event.

I think the disconnect is always an error situation which is not expected.

Reply:
The events are not used anywhere else except for printing if I read the code correctly.

I'm always concerned about unjoined/unterminated threads - imagine `event_subscriber` thread can't connect to parachain, will it get terminated? If it's always live, then we'll have more and more detached threads.

I'm OK to apply the automatic reconnection only if we can manage the lifecycle of the thread (e.g. we guarantee there's maximum one such subscriber thread at any time)